src/fmu/dataio/scripts/copy_preprocessed.py

```diff
@@ -0,0 +1,167 @@
+
+This script is intended to be run through an ERT HOOK PRESIM workflow.
+
+"""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import warnings
+from pathlib import Path
+from typing import TYPE_CHECKING, Final
+
+from fmu.dataio import ExportPreprocessedData
+
+try:
+    from ert.shared.plugins.plugin_manager import hook_implementation
+except ModuleNotFoundError:
+    from ert_shared.plugins.plugin_manager import hook_implementation
+
+try:
+    from ert.config import ErtScript
+except ImportError:
+    from res.job_queue import ErtScript
+
+if TYPE_CHECKING:
+    from ert.shared.plugins.workflow_config import WorkflowConfigs
+
+logger: Final = logging.getLogger(__name__)
+logger.setLevel(logging.CRITICAL)
+
+# This documentation is compiled into ert's internal docs
+DESCRIPTION = """
+WF_COPY_PREPROCESSED_DATAIO will copy preprocessed data to a FMU run at
+<caseroot>/share/observations/. If the data contains metadata this will be
+updated with information about the FMU run and ready for upload to sumo.
+
+Preprocessed data refers to data that has been exported with dataio outside of a FMU
+context, and is typically located in a share/preprocessed/ folder on the project disk.
+"""
+
+EXAMPLES = """
+Create an ERT workflow e.g. named ``ert/bin/workflows/xhook_copy_preprocessed_data``
+with the contents::
+  WF_COPY_PREPROCESSED_DATAIO <SCRATCH>/<USER>/<CASE_DIR> <CONFIG_PATH> '../../share/preprocessed/'
+
+Add following lines to your ERT config to have the job automatically executed::
+  LOAD_WORKFLOW ../bin/workflows/xhook_copy_preprocessed_data
+  HOOK_WORKFLOW xhook_copy_preprocessed_data PRE_SIMULATION
+"""  # noqa
+
+
+def main() -> None:
+    """Entry point from command line
+
+    When script is called from an ERT workflow, it will be called through the 'run'
+    method on the WfCopyPreprocessedData class. This context is the intended usage.
+    The command line entry point is still included, to clarify the difference and
+    for debugging purposes.
+    """
+    parser = get_parser()
+    commandline_args = parser.parse_args()
+    copy_preprocessed_data_main(commandline_args)
+
+
+class WfCopyPreprocessedData(ErtScript):
+    """A class with a run() function that can be registered as an ERT plugin.
+
+    This is used for the ERT workflow context. It is prefixed 'Wf' to avoid a
+    potential naming collisions in fmu-dataio."""
+
+    def run(self, *args: str) -> None:
+        """Parse arguments and call copy_preprocessed_data_main()"""
+        parser = get_parser()
+        workflow_args = parser.parse_args(args)
+        copy_preprocessed_data_main(workflow_args)
+
+
+def copy_preprocessed_data_main(args: argparse.Namespace) -> None:
+    """Copy the preprocessed data to scratch and upload it to sumo."""
+
+    check_arguments(args)
+    logger.setLevel(args.verbosity)
+
+    searchpath = Path(args.ert_config_path) / args.inpath
+    match_pattern = "[!.]*"  # ignore metafiles (starts with '.')
```

Comment:
Is there any possibility this ignores other files we want? (I assume not: but just a sanity check)